pkg/api/signing_cert.go

```diff
@@ -31,7 +34,7 @@ import (
 	"github.com/sigstore/fulcio/pkg/ctl"
 	"github.com/sigstore/fulcio/pkg/generated/restapi/operations"
 	"github.com/sigstore/fulcio/pkg/log"
-	"github.com/sigstore/fulcio/pkg/oauthflow"
```

Comment:
It would be nice if we could make this modular. I feel like this is locking into one provider and we are going to have to refactor it at a later point to support other CAs.

Reply:
This PR doesn't actually change or add the CA integration, it just refactors it a bit. Agree we should add some kind of intermediate layer.

Reply:
It's a fair point in that there is no other CA right now to have a modular layer, I will open an issue for this.

Reply:
Thanks!